x/bank/keeper/msg_server.go

```diff
@@ -27,14 +27,20 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 }
 
 func (k msgServer) Send(goCtx context.Context, msg *types.MsgSend) (*types.MsgSendResponse, error) {
-	from, err := sdk.AccAddressFromBech32(msg.FromAddress)
-	if err != nil {
-		return nil, sdkerrors.ErrInvalidAddress.Wrapf("invalid from address: %s", err)
-	}
-
-	to, err := sdk.AccAddressFromBech32(msg.ToAddress)
-	if err != nil {
-		return nil, sdkerrors.ErrInvalidAddress.Wrapf("invalid to address: %s", err)
+	var (
+		from, to []byte
+		err      error
+	)
+
+	if base, ok := k.Keeper.(BaseKeeper); ok {
+		from, err = base.ak.AddressCodec().StringToBytes(msg.FromAddress)
+		if err != nil {
+			return nil, sdkerrors.ErrInvalidAddress.Wrapf("invalid from address: %s", err)
+		}
+		to, err = base.ak.AddressCodec().StringToBytes(msg.ToAddress)
+		if err != nil {
+			return nil, sdkerrors.ErrInvalidAddress.Wrapf("invalid to address: %s", err)
+		}
 	}
```

Comment:
We should handle the case where the keeper isn't a `BaseKeeper`? Otherwise can from and to will be both empty.